### PR TITLE
feat(YouTube): add UI to disable privacy mode on a per-video basis

### PR DESCRIPTION
**Pull request by veryCrunchy**
My brain be hurting now..

Adds a button to overwrite the users privacy settings
Overwrites are per privacy mode (on/off) and not global. Overwrites have a 3 hour ttl, ttl gets checked on page load,

https://github.com/PreMiD/Presences/assets/73395916/bfcf92b9-55ac-46bd-8d8b-a66711674831

## Screenshots
<details>

image
image

</details>

### CODE REVIEW

**Review by theusaf — COMMENTED** [review]
Maybe move the code for this to a separate file to keep the main script less messy?

**Comment by veryCrunchy** [comment]
> Maybe move the code for this to a separate file to keep the main script less messy?

Torture..

**Comment by veryCrunchy** [comment]
I believe that should do it

**Comment by veryCrunchy** [comment]
I think people are going to love it.
I already can't go without 😭

**Review by Bas950 — CHANGES_REQUESTED** [review]
Move it to a util/pvPrimary.ts or something like that

**Comment by joerkig** [comment]
Would it also be possible to have an option to disable showing the button?

**Comment by veryCrunchy** [comment]
@joerkig I can add a setting for that if you want me to?

**Comment by joerkig** [comment]
> @joerkig I can add a setting for that if you want me to?

Would be nice, cause me personally I don't think I will use this that much and would be appreciated to not have extra clutter in that case

**Comment by veryCrunchy** [comment]
Donesies

File: websites/Y/YouTube/util/pvPrivacyUI.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+);
+
+export function pvPrivacyUI(
+	privacy: boolean,
+	videoId: string,
+	privacyTtl: number
+): boolean {
+	let perVideoPrivacy = false;
+	const isVideoInArray = (videoId: string, array: VPArray) => {
+		return array.findIndex(entry => entry.videoId === videoId) !== -1;
```
Comment (Bas950) [inline websites/Y/YouTube/util/pvPrivacyUI.ts:28]:
```suggestion
		return array.some(entry => entry.videoId === videoId);
```

File: websites/Y/YouTube/presence.ts [hunk 2]
```diff
@@ -182,11 +186,26 @@ presence.on("UpdateData", async () => {
...
+				? (perVideoPrivacy = pvPrivacyUI(
+						privacy,
+						new URLSearchParams(search).get("v"),
+						privacyTtl
+				  ))
+				: document.querySelector("#pmdEnablePrivacy")
+				? (document.querySelector("#pmdEnablePrivacy").remove(),
+				  document.querySelector("#pmdEnablePrivacyTooltip").remove())
+				: "";
+		}
```
Comment (Bas950) [inline websites/Y/YouTube/presence.ts:201]: probs and if-else is more readable here

File: websites/Y/YouTube/util/pvPrivacyUI.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+				perVideoPrivacy = !isVideoInArray(videoId, perVideoNonPrivacyArray);
+				localStorage.setItem("pmdPrivacyEnabled", "true");
+			} else {
+				perVideoPrivacy = isVideoInArray(videoId, perVideoPrivacyArray);
+				localStorage.setItem("pmdPrivacyEnabled", "false");
+			}
+
+			const svgEnabled = document.querySelector(
+					"#pmdPrivacyEnabled"
+				) as HTMLElement,
```
Comment (Bas950) [inline websites/Y/YouTube/util/pvPrivacyUI.ts:121]:
```suggestion
			const svgEnabled = document.querySelector<HTMLElement>(
					"#pmdPrivacyEnabled"
				),
```

File: websites/Y/YouTube/util/pvPrivacyUI.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+				svgDisabled = document.querySelector(
+					"#pmdPrivacyDisabled"
+				) as HTMLElement;
```
Comment (Bas950) [inline websites/Y/YouTube/util/pvPrivacyUI.ts:124]:
```suggestion
				svgDisabled = document.querySelector<HTMLElement>(
					"#pmdPrivacyDisabled"
				);
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by theusaf** [comment]
Can you also provide a screenshot of the [settings](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#reviewing)?

**Comment by veryCrunchy** [comment]
<details><summary>Screenshots</summary>
<p>

image
image

</p>
</details>

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**